src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java

```diff
@@ -0,0 +1,256 @@
+        this.errorMessage = "Cannot start Artemis. Please start the release " + requiredVersion + " first, otherwise the migration will fail";
+    }
+}
+
+/**
+ * Handles the database migration process by checking against defined migration paths to ensure
+ * that the database schema is up-to-date before application startup. This class plays a critical
+ * role in optimizing application startup time by merging Liquibase migrations and eliminating Java
+ * migrations for each major version release.
+ * <p>
+ * This mechanism is essential for maintainers, aiming to streamline the startup, especially
+ * beneficial for server and e2e tests. It mandates administrators to upgrade to the latest minor
+ * version before a major update, facilitating a smooth transition and incorporation of all
+ * database changes.
+ * <p>
+ * Steps for new database migration paths:
+ * <ol>
+ * <li>Re-create the new initial scheme from the database after applying all existing liquibase migrations.</li>
+ * <li>Delete all existing liquibase migrations except the new initial scheme and *_cleanup.xml (adapt *_cleanup.xml with the latest date).</li>
+ * <li>Delete all Java migrations that have been executed before.</li>
+ * <li>Check that the new initial scheme is compatible with MySQL and Postgres and can be started from scratch in both environments.</li>
+ * <li>Add the new migration path in the constructor of DatabaseMigration.</li>
+ * <li>Verify that the migration works with the designated path (ideally with a local database and with a dump from a test / production system).</li>
+ * <li>Document the migration carefully in the release notes and the online documentation.</li>
+ * </ol>
+ */
+public class DatabaseMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(DatabaseMigration.class);
+
+    private final DataSource dataSource;
+
+    private final List<MigrationPath> migrationPaths = new ArrayList<>();
+
+    private final String currentVersionString;
+
+    private String previousVersionString;
+
+    public DatabaseMigration(String currentVersionString, DataSource dataSource) {
+        this.currentVersionString = currentVersionString;
+        this.dataSource = dataSource;
+
+        // Initialize migration paths here in the correct order
+        migrationPaths.add(new MigrationPath("5.12.9")); // needed between 6.0.0 and 7.0.0
+        migrationPaths.add(new MigrationPath("6.9.6")); // needed between 7.0.0 and 8.0.0
+
+        // Add more migrations here as needed
+    }
+
+    public String getPreviousVersionString() {
+        return previousVersionString;
+    }
+
+    /**
+     * Checks against the defined migration paths to determine if the current version of the database
+     * is compatible or requires migration. This method ensures that the database schema is prepared
+     * and up-to-date before proceeding with the application startup.
+     */
+    public void checkMigrationPath() {
```

Comment:
Feels like this method could be static and take the current version and datasource.
The migration paths could just be a static final list.

Then we could simply do the check in the `LiquibaseConfiguration`: `DatabaseMigration.checkMigrationPath(currentVersion, datasource);`

What do you think?